packages/extension-browser/src/devtools/utils/activity.ts

```diff
@@ -0,0 +1,79 @@
+export type Activity = {
+    [key in keyof {
+
+        /**
+         * 28-entry list of 1 (active) and 0 (inactive) days.
+         * (e.g. `"1001100110011001100110011001"`)
+         */
+        last28Days: string;
+
+        /**
+         * Date in ISO 8601 format (toISOString) with time at zero.
+         * (e.g. `"2019-10-04T00:00:00.000Z"`)
+         */
+        lastUpdated: string;
+
+    }]: string;
+};
+
+const getISODateString = () => {
+    const date = new Date(Date.now());
+
+    date.setUTCHours(0);
+    date.setUTCMinutes(0);
+    date.setUTCSeconds(0);
+    date.setUTCMilliseconds(0);
+
+    return date.toISOString();
+};
+
+const getDaysBetweenUpdates = (currentUpdate: string, lastUpdated?: string) => {
+    if (!lastUpdated) {
+        return 1;
+    }
+
+    const deltaMS = new Date(currentUpdate).getTime() - new Date(lastUpdated).getTime();
+
+    return deltaMS / 1000 / 60 / 60 / 24;
+};
+
+const initializeActivity = (): Activity => {
+    const lastUpdated = '';
+    const last28Days = new Array(28)
```

Comment:
what about `''.padEnd(28, 0)` instead of the `new Array` thing?

Reply:
😍